ddclient.in

```diff
@@ -2146,6 +2154,128 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## geturl -- via curl
+######################################################################
+sub curl_cmd {
+    my $params = join("\n",@_);
+    my $tmpfile;
+
+    return '' if (length($params) == 0); ## no parameters provided
+
+    # debug("CURL:\r\n%s\r\n",$params); ## Uncomment only if required as may expose credentials
+
+    eval "no warnings 'all'; use File::Spec;";
+    $tmpfile = ((!$@) ? File::Spec->tmpdir() : '/tmp').'/ddclient-';
+    my @chars = ("A".."Z", "a".."z", "0".."9");
+    $tmpfile .= $chars[rand @chars] for 1..10;
+    debug("CURL Tempfile: %s",$tmpfile);
+    open(my $TFH, '>', $tmpfile) or fatal("Falied to open $tmpfile for curl");
+    chmod(0600,$tmpfile);
+    print($TFH $params);
+    close($TFH);
+    my $reply = qx{ curl --config $tmpfile 2>/dev/null };
```

Comment:
Shell metacharacters in `$tmpfile` need to be quoted. Or use `open -|` with a list.

Reply:
Do you mean in the filename itself, or in the contents of the file?  For the contents, we're escaping double quotes before passing to this function.

Reply:
The filename itself.

Reply:
We are generating the filename ourself... combination of dir path from the system plus a filename built from `_A-Za-z0-9`.  I don't think there can be any metacharacters that we need to worry about?

Reply:
I have single quoted `'$tmpfile'` which should handle it.

Reply:
Unless a malicious actor puts a single quote in `$ENV{TMPDIR}`.

Reply:
Should be safe now.  Have hardcoded /tmp and we generate our own filename.